src/node_crypto.cc

```diff
@@ -3105,23 +3113,69 @@ void CipherBase::InitIv(const FunctionCallbackInfo<Value>& args) {
   ASSIGN_OR_RETURN_UNWRAP(&cipher, args.Holder());
   Environment* env = cipher->env();
 
-  CHECK_GE(args.Length(), 3);
+  CHECK_GE(args.Length(), 4);
 
   const node::Utf8Value cipher_type(env->isolate(), args[0]);
   ssize_t key_len = Buffer::Length(args[1]);
   const char* key_buf = Buffer::Data(args[1]);
   ssize_t iv_len = Buffer::Length(args[2]);
   const char* iv_buf = Buffer::Data(args[2]);
-  cipher->InitIv(*cipher_type, key_buf, key_len, iv_buf, iv_len);
+  CHECK(args[3]->IsInt32());
+  // Don't assign to cipher->auth_tag_len_ directly; the value might not
+  // represent a valid length at this point.
+  int auth_tag_len = args[3]->Int32Value();
+
+  cipher->InitIv(*cipher_type, key_buf, key_len, iv_buf, iv_len, auth_tag_len);
+}
+
+
+bool CipherBase::InitAuthenticated(const char *cipher_type, int iv_len,
+                                   int auth_tag_len) {
+  // TODO(tniessen) Use EVP_CTRL_AEAD_SET_IVLEN when migrating to OpenSSL 1.1.0
```

Comment:
this needs to be handled, we support 1.1.0 now and our CI tests against it

Reply:
These constants still exist in OpenSSL 1.1.0 and `EVP_CTRL_AEAD_SET_IVLEN` is an alias for `EVP_CTRL_GCM_SET_IVLEN` which should be the same as `EVP_CTRL_CCM_SET_IVLEN`, so this is technically still correct. My intention was to use the `EVP_CTRL_AEAD_*` constants once we don't support 1.0.2 anymore.